account_check_printing_report_base/tests/test_check_printing_report.py

```diff
@@ -0,0 +1,157 @@
+# © 2016 Eficent Business and IT Consulting Services S.L.
+# Copyright 2018 iterativo.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import time
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestAccountCheckPrintingReportBase(TransactionCase):
+    def setUp(self):
+        super(TestAccountCheckPrintingReportBase, self).setUp()
+        self.account_move_model = self.env["account.move"]
+        self.journal_model = self.env["account.journal"]
+        self.register_payments_model = self.env["account.payment.register"]
+        self.payment_method_model = self.env["account.payment.method"]
+        self.account_move_line_model = self.env["account.move.line"]
+        self.account_account_model = self.env["account.account"]
+        self.payment_model = self.env["account.payment"]
+
+        self.partner1 = self.env.ref("base.res_partner_1")
+        self.company = self.env.ref("base.main_company")
+        self.currency_usd_id = self.env.ref("base.USD").id
+        self.currency_euro_id = self.env.ref("base.EUR").id
+        self.acc_payable = self.env.ref("account.data_account_type_payable")
+        self.acc_expense = self.env.ref("account.data_account_type_expenses")
+        self.product = self.env.ref("product.product_product_4")
```

Comment:
You should create test data instead of using demo data. That way the test will be more resilient and independent from previous operations on the DB

Reply:
In this way I use the test created on the previous version, I usually make what you say. 

Probably the best way is change the declaration to avoid what you say, but doing like is done we re-use the code from previous versions and I think that this solution could be repetitive, because I will define at that recorsets the same data that is declared yet on demo data. What do you think?

Reply:
Well, there is no problem in creating new data just for the test, even if similar records already exist as demo. By doing this, you make sure your tests won't fail just because a test from a previous model changed or deleted the record you are trying to use. 

However, if the code already comes from the previous version and is working, maybe somebody else could give their opinion on if it is necessity to re-write. @carlosdauden @sergio-teruel